sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -67,17 +67,16 @@ class KernelProgramCache {
   using PiProgramT = std::remove_pointer<RT::PiProgram>::type;
   using PiProgramPtrT = std::atomic<PiProgramT *>;
   using ProgramWithBuildStateT = BuildResult<PiProgramT>;
-  using ProgramCacheKeyT =
-      std::pair<std::pair<SerializedObj, KernelSetId>, RT::PiDevice>;
+  using ProgramCacheKeyT = std::pair<std::pair<SerializedObj, KernelSetId>,
+                                     std::pair<RT::PiDevice, string_class>>;
   using ProgramCacheT = std::map<ProgramCacheKeyT, ProgramWithBuildStateT>;
   using ContextPtr = context_impl *;
 
   using PiKernelT = std::remove_pointer<RT::PiKernel>::type;
 
   using PiKernelPtrT = std::atomic<PiKernelT *>;
   using KernelWithBuildStateT = BuildResult<PiKernelT>;
-  using KernelByNameT =
-      std::map<std::pair<string_class, RT::PiDevice>, KernelWithBuildStateT>;
+  using KernelByNameT = std::map<string_class, KernelWithBuildStateT>;
```

Comment:
`RT::PiDevice` was added to this map to support context with multiple devices. @vladimirlaz, am I understood correctly that removal of device id from the map doesn't affect this case?

Reply:
It was was discussed in design review:

sergey-semenov 28 days ago Member
Seems to me that both the device and build options components are redundant here, since those are associated with the program which is already a component of the key by itself. Am I missing something here?

kbobrovs 28 days ago Member
+1. It is not possible to compile individual kernel from a program with different options or for a different device.

Reply:
Thanks.